File: xamarin/android/PropertyCross/Views/FavouritesView.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using Android.App;
using Android.Content;
using Android.Content.PM;
using Android.OS;
using Android.Runtime;
using Android.Views;
using Android.Widget;
using PropertyCross.Presenter;
using PropertyCross.Model;
using Com.Actionbarsherlock.App;
using Com.Actionbarsherlock.View;

using IMenuItem = global::Com.Actionbarsherlock.View.IMenuItem;

namespace com.propertycross.xamarin.android.Views
{
	[Activity (ScreenOrientation = ScreenOrientation.Portrait)]			
	public class FavouritesView : SherlockListActivity, FavouritesPresenter.View
	{		
		private FavouritesPresenter presenter;

		protected override void OnCreate(Bundle bundle)
		{
			base.OnCreate(bundle);

			SupportActionBar.Title = Resources.GetString(Resource.String.favourites_view);
			SupportActionBar.SetDisplayHomeAsUpEnabled(true);

			ListAdapter = new SearchResultsAdapter(this, new List<Property>() {});
			
			var app = PropertyCrossApplication.GetApplication(this);
			presenter = (FavouritesPresenter) app.Presenter;
			app.CurrentActivity = this;
		}

		protected override void OnResume()
		{
			base.OnResume();
			if (presenter != null)
			{
				presenter.SetView(this);
			}
		}

		public override bool OnOptionsItemSelected(IMenuItem item)
		{
			if(item.ItemId == Android.Resource.Id.Home)
			{
				Finish();
				return true;
			}
			return base.OnOptionsItemSelected(item);
		}

		protected override void OnListItemClick(ListView l, View v, int position, long id)
		{
			var adapter = (SearchResultsAdapter) ListAdapter;
			Property item = adapter[position];
			PropertySelected(this, new PropertyEventArgs(item));
		}

		public void SetFavourites(List<Property> properties)
		{
			ListAdapter = new SearchResultsAdapter(this, properties);
		}
		
		public event EventHandler<PropertyEventArgs> PropertySelected;
	}
}

File: native/android/PropertyCross/src/com/propertycross/android/views/FavouritesView.java
package com.propertycross.android.views;

import java.util.ArrayList;
import java.util.List;

import android.os.Bundle;
import android.view.View;
import android.widget.ListView;

import com.actionbarsherlock.app.SherlockListActivity;
import com.actionbarsherlock.view.MenuItem;
import com.propertycross.android.R;
import com.propertycross.android.events.Callback;
import com.propertycross.android.events.PropertyEventArgs;
import com.propertycross.android.events.PropertySelectedEvent;
import com.propertycross.android.model.Property;
import com.propertycross.android.presenter.FavouritesPresenter;

public class FavouritesView
	extends SherlockListActivity
	implements FavouritesPresenter.View {

	private FavouritesPresenter presenter;
	private Callback<PropertySelectedEvent> propertySelectedCallback;
	
	@Override
	public void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		
		getSupportActionBar().setTitle(
				getResources().getString(R.string.favourites_view));
		getSupportActionBar().setDisplayHomeAsUpEnabled(true);
		
		setListAdapter(new SearchResultsAdapter(this, new ArrayList<Property>()));
		
		PropertyFinderApplication app = PropertyFinderApplication.getApplication(this);
		presenter = (FavouritesPresenter) app.presenter;
		presenter.setView(this);
		app.currentActivity = this;		
	}
	
	@Override
	public boolean onOptionsItemSelected(MenuItem item) {
		if (item.getItemId() == android.R.id.home) {
			finish();
			return true;
		}
		
		return super.onOptionsItemSelected(item);
	}
	
	@Override
	protected void onListItemClick(ListView l, View v, int position, long id) {
		Property item = ((SearchResultsAdapter) getListAdapter()).getItem(position);
		if (propertySelectedCallback != null) {
			propertySelectedCallback.complete(
					new PropertySelectedEvent(this, new PropertyEventArgs(item)));
		}
	}

	@Override
	public void setFavourites(List<Property> properties) {
		if(properties != null) {
			((SearchResultsAdapter) getListAdapter()).addRange(properties);
		}
	}

	@Override
	public void propertySelected(Callback<PropertySelectedEvent> callback) {
		propertySelectedCallback = callback;
	}	
}
